### PR TITLE
Chore/clean up yarn cache

### PR DESCRIPTION
**Pull request by mimse**
Removed our local cache to reduce the size of the repository. They were not used by the CI
Also updated to the newest node action, and uses its yarn cache function instead of our "own"

### CODE REVIEW

File: .github/workflows/detect-changes.yml [hunk 1]
```diff
@@ -78,12 +78,11 @@ jobs:
...
       - name: Set up Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version-file: .nvmrc
-
-      - name: 📥 Monorepo install
-        uses: ./.github/actions/yarn-nm-install
+          cache: yarn
+      - run: yarn install --immutable
```
Comment (dzhavat) [inline .github/workflows/detect-changes.yml:85]: suggestion: Let's give this task a `name` so it appears with something more meaningful in the UI. For example "Install dependencies"

image

File: .github/workflows/dh-ci-frontend.yml [hunk 2]
```diff
@@ -40,12 +40,11 @@ jobs:
...
       - name: Set up Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version-file: ${{ env.NODE_VERSION_FILE }}
-
-      - name: 📥 Monorepo install
-        uses: ./.github/actions/yarn-nm-install
+          cache: yarn
+      - run: yarn install --immutable
```
Comment (dzhavat) [inline .github/workflows/dh-ci-frontend.yml:47]: suggestion: let's give this task a `name`. The comment applies to all `yarn install` tasks bellow.

Comment (mimse) [reply .github/workflows/dh-ci-frontend.yml]: Good idea

**Review by dzhavat — APPROVED** [review]
praise: nice changes. Let's try it 💯

**Review by Renetnielsen — APPROVED** [review]

**Pull request merged**